hil/ext/switches/juniper.py

```diff
@@ -0,0 +1,373 @@
+        """ Clears the values of the following variables."""
+        self.enable_interface = None
+        self.set_trunk_mode = None
+        self.add_vlan = None
+        self.remove_vlan = None
+        self.set_native_vlan = None
+        self.remove_native_vlan = None
+        self.remove_default_vlan = None
+
+    def _set_load_config(self, o_session, set_command):
+        """ Load configuration changes using the 'set' method.
+        `o_session`: session object with connection to switch already open.
+        `set_command`: Multi-line `set` method based command string.
+        """
+        jun = o_session
+        try:
+            jun.cfg.load(set_command, format="set")
+        except ConfigLoadError as e:
+            if e.message == "warning: statement not found":
+                pass
+            else:
+                return e
+
+    def _jinja_load_config(self, o_session, j_template, var_dict):
+        """ Load configuration changes using the 'jinja_template' method.
+        `o_session`: session object with connection to switch already open.
+       `j_template`: jinja2 compliant configuration template. Samples are
+       available at `junos/jinja_templates/`.
+       `var_dict`: Dictionary of variables to be used in the jinja template
+       passed with `j_template`, that will load the specific configuration
+       change in the switch.
+       """
+        jun = o_session
+        jun.cfg.load(
+               template_path=j_template, template_vars=var_dict,
+               format='text'
+               )
+
+    def _commit_config(self, loaded_session):
+        """ Takes an open session that has loaded configuration change.
+        Commits it if it is correct, otherwise rollback
+        to recent commited state.
+        `loaded_session`: session object after new configuration is loaded.
+        """
+        jun = loaded_session
+        if jun.cfg.commit_check:
+            jun.cfg.commit()
+        else:
+            jun.cfg.rollback()
+            raise ConfigCommitError
+
+    def _set_commit_config(self, set_command):
+        """ Changes the switch config by using only 'set' method.
+        `set_command`: 'set' method based command string.
+        Note: Some configurations changes require both 'set' and
+        'jinja_template' method. In those cases use combination of functions
+        `_set_load_config`, `_jinja_load_config` as needed
+        finally commit using `_commit_config`.
+        """
+
+        jun = self._create_session()
+        try:
+            with jun:
+                self._set_load_config(jun, set_command)
+                try:
+                    self._commit_config(jun)
+                except ConfigCommitError as err:
+                    message = (
+                            "Inconsistent configuration, \
+                                    cannot commit. Exiting.".format(err)
+                                    )
+                    return sys.exit(1)
```

Comment:
return sys.exit(1)? Why are you returning this? Can't you just exit here?

Reply:
my bad. fixed it.